### DISCUSSION TITLE
Simulate Connected Autonomous Vehicles

### DISCUSSION THREAD

**Discussion by Lplenka** [eleurent/highway-env#311] [General]
Hey @eleurent  

Thanks for creating this amazing project.

I have used the "highway-fast-v0"  environment with the intention to work on Deep Reinforcement Learning for autonomous lane changing. Though this environment simulates autonomous vehicles quite well I am not sure if this environment can also be used for Connected Autonomous Vehicles (CAV). Is it justified to use this to simulate lane changing for CAVs?

Is there any part of the code I can update to add CAV functionality?

I am a beginner and any suggestion on the right choice of environment would be very helpful. Thanks in advance.

**Comment by eleurent** [comment]
Hi! No, unfortunately I haven't work with of connected vehicles specifically before.
There is however the option to have multiple controlled vehicles in a scene, and to get the joint observation (tuple of per-vehicle observations), that we can use for centralized decision-making (or decentralized decision making but with shared information) before returning the tuple of their individual actions.
So this mechanism can probably adapted to suit your use case.

**Comment by Lplenka** [comment]
Thanks for the response @eleurent 

If I understand  this correctly, then every automated vehicle in the environment is getting joint observations of all surrounding vehicles (human driven and automated vehicles). Based on this shared information the ego vehicle will take the decision.


This can be considered as a multi-agent simulation of Connected Autonomous Vehicles as in real world a CAV will get individual information from each surrounding CAV through internet or bluetooth. These independent information when combined by that CAV is similar to the shared information automated vehicles  are getting in this environment. 

Hence, using this shared information from the environment, I can use multi-agent reinforcement learning  for lane changing tasks for CAVs.

Hopefully, this analogy will work.

**Reply by eleurent** [reply]
More precisely, only automated vehicles have "sensors" and provide local measurements as part of the joint observation, not "human" (uncontrolled) vehicles.

**Reply by Lplenka** [reply]
Yes, that makes sense. Thanks a lot :)
